File: app/Models/FarmerCommodity.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class FarmerCommodity extends Model
{
    use HasFactory, SoftDeletes;

    protected $table = 'farmer_commodities';

    protected $fillable = [
        'user_id',
        'name',
        'code',
        'unit',
        'description',
        'status',
    ];

    /**
     * Scope query to only active commodities.
     */
    public function scopeActive($query)
    {
        return $query->where('status', 'active');
    }

    /**
     * Scope query for a specific farmer.
     */
    public function scopeForUser($query, int $userId)
    {
        return $query->where('user_id', $userId);
    }

    /**
     * Owner (farmer) of the commodity.
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    /**
     * Harvests associated with this commodity.
     */
    public function harvests(): HasMany
    {
        return $this->hasMany(Harvest::class, 'commodity_id');
    }

    /**
     * Planting seasons associated with this commodity.
     */
    public function seasons(): HasMany
    {
        return $this->hasMany(Season::class, 'commodity_id');
    }
}

File: mobile_app/lib/models/farmer_commodity.dart
class FarmerCommodity {
  final int id;
  final int userId;
  final String name;
  final String? code;
  final String unit;
  final String? description;
  final String status;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final Map<String, dynamic>? farmer;

  FarmerCommodity({
    required this.id,
    required this.userId,
    required this.name,
    this.code,
    required this.unit,
    this.description,
    required this.status,
    this.createdAt,
    this.updatedAt,
    this.farmer,
  });

  bool get isActive => status == 'active';

  factory FarmerCommodity.fromJson(Map<String, dynamic> json) {
    return FarmerCommodity(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id']?.toString() ?? '0') ?? 0,
      userId: json['user_id'] is int ? json['user_id'] : int.tryParse(json['user_id']?.toString() ?? '0') ?? 0,
      name: json['name']?.toString() ?? '',
      code: json['code']?.toString(),
      unit: json['unit']?.toString() ?? 'kg',
      description: json['description']?.toString(),
      status: json['status']?.toString() ?? 'active',
      createdAt: json['created_at'] != null ? DateTime.tryParse(json['created_at'].toString()) : null,
      updatedAt: json['updated_at'] != null ? DateTime.tryParse(json['updated_at'].toString()) : null,
      farmer: json['farmer'] is Map<String, dynamic> ? json['farmer'] as Map<String, dynamic> : null,
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'user_id': userId,
    'name': name,
    'code': code,
    'unit': unit,
    'description': description,
    'status': status,
  };
}
